File: MergeIntervals.py
class Interval(object):
	"""docstring for Interval"""
	def __init__(self, s=0, e=0):
		self.start = s
		self.end = e

class Solution(object):
	"""docstring for Solution"""
	def merge(self, intervals):
		intervals.sort(key = lambda x:x.start)
		results = []
		if len(intervals) == 0:
			return []

		tmp = Interval(intervals[0].start, intervals[0].end)
		for i in range(1, len(intervals)):
			next = intervals[i]
			if tmp.end >= next.end:
				pass
			elif tmp.end >= next.start:
				tmp.end = next.end
			elif tmp.end < next.start:
				results.append(Interval(tmp.start, tmp.end))
				# update
				tmp.start = next.start
				tmp.end = next.end
		results.append(tmp)
		print(results[0].start)
		return results



interval1 = Interval(1, 4)
interval2 = Interval(5, 6)
interval3 = Interval(8, 10)
interval4 = Interval(15, 18)

solution = Solution()
results = solution.merge([interval1, interval2])
print(len(results))

File: MergeIntervals.java
package com.company;

import java.lang.reflect.Array;
import java.sql.SQLSyntaxErrorException;
import java.util.ArrayList;
import java.util.*;
import java.util.Queue;
import java.util.regex.Matcher;

/**
 * Created by yiqin on 9/10/15.
 */
public class MergeIntervals {

    public static void main(String[] args) {

        List<Interval> intervals = new ArrayList<Interval>();

        Interval interval1 = new Interval(1, 3);
        Interval interval2 = new Interval(2, 6);
        Interval interval3 = new Interval(8, 10);
        Interval interval4 = new Interval(15, 18);

        intervals.add(interval1);
        intervals.add(interval2);
        intervals.add(interval3);
        intervals.add(interval4);


        merge(intervals);

    }

    public static List<Interval> merge(List<Interval> intervals) {








        return intervals;
    }

}
